Implementation of toggling layer visibility

srmanc opened the pull request:
This PR addresses toggling layer visibility when a correct layer ID is supplied.

Works cross-platform. Implementation in example project provided as well.

felix-ht commented on lib/src/controller.dart:
```diff
@@ -1130,6 +1130,10 @@ class MapboxMapController extends ChangeNotifier {
     return _mapboxGlPlatform.setFilter(layerId, filter);
   }
 
+  Future<void> setVisibility(String layerId, bool isVisible) {
```
Might be a good idea to add some documentation here.

felix-ht commented on mapbox_gl_web/lib/src/mapbox_web_gl_platform.dart:
```diff
@@ -695,6 +695,12 @@ class MapboxWebGlPlatform extends MapboxGlPlatform
 
+  @override
+  Future<void> setVisibility(String layerId, bool isVisible) async {
+    _map.setLayoutProperty(
```
Would this throw if the layer does not exist?
If so this would be the only platform were one gets in this situation right?

srmanc commented:
Thanks. Added requested changed in https://github.com/flutter-mapbox-gl/maps/pull/1197/commits/74387117100a4bd19f93758bebed315aa2fb1b4d. 

> Would this throw if the layer does not exist?
If so this would be the only platform were this would be the case right?

It does not throw, however it leaves a 'null' message in the debug window, which is rather annoying, so I added the check anyway.

srmanc commented:
Fixed linter errors as well.

carson1998 commented:
im in need of this feature. Hopefully we can get this update soon

srmanc commented:
I think this is ready to be merged.

carson1998 commented:
I'm seeing some failed checks for this change request. What seems to be the problem?

srmanc commented:
No failed checks. This repo's CI fails to build artifacts because of incorrectly wired secrets / tokens. This is ready to be merged. @felix-ht

felix-ht commented:
@srmanc the checks actually failed as of https://github.com/flutter-mapbox-gl/maps/pull/1212

felix-ht commented:
@srmanc could you rebase on master?

srmanc commented:
@felix-ht You're right, didn't see that. Just rebased.

felix-ht commented:
@srmanc  i think something went wrong with the merge - (cherry pick?) - as this shows a large number of changes (same with the other pr)

srmanc commented:
@felix-ht I just did rebase on master and then a pull --rebase and a push, somehow changes did not reflect the master tip. I did another merge with master and pushed, seems okay now.

carson1998 commented:
still pending huh.

srmanc commented:
@felix-ht Do you think we could look at all the pending PRs in the near future? @carson1998

connybruhn commented:
I'm also waiting for this important feature. I need it for a radar player in my weather app.
Is there a chance to get that feature this year? Otherwise I would have to switch back to another map package.

connybruhn commented:
Any news here? I would really like to use this package, but I need that feature.

srmanc commented:
Ping.

felix-ht reviewed (approved):
LGTM

dhaq commented:
When can I use this toggle layer visibility feature?
